src/groth16/prover.rs

```diff
@@ -181,6 +182,8 @@ where
     E: Engine,
     C: Circuit<E>,
 {
+    let lck = gpu::lock().expect("Cannot obtain prover lock!");
```

Comment:
Should this be a panic? Or should it be an error? (I don't really know how this code is called from the outside, so I'm really just asking :)

Reply:
@vmx Better to be an error, I'm going to remove these panics in a separate PR.